samples/stock/demo/all-indicators/demo.js

```diff
@@ -0,0 +1,193 @@
+Highcharts.getJSON('https://demo-live-data.highcharts.com/aapl-ohlcv.json', function (data) {
+    // split the data set into ohlc and volume
+    var ohlc = [],
+        volume = [],
+        dataLength = data.length,
+
+        i = 0;
+
+    for (i; i < dataLength; i += 1) {
+        ohlc.push([
+            data[i][0], // the date
+            data[i][1], // open
+            data[i][2], // high
+            data[i][3], // low
+            data[i][4] // close
+        ]);
+
+        volume.push([
+            data[i][0], // the date
+            data[i][5] // the volume
+        ]);
+    }
+
+    // create the chart
+    const chart = Highcharts.stockChart('container', {
+        chart: {
+            height: 600
+        },
+        title: {
+            text: 'AAPL Historical'
+        },
+        subTitle: {
+            text: 'All indicators'
+        },
+        legend: {
+            enabled: true
+        },
+        rangeSelector: {
+            selected: 2
+        },
+        tooltip: {
+            split: true
+        },
+        yAxis: [{
+            height: '70%'
+        }, {
+            top: '75%',
+            height: '25%'
+        }],
+        series: [{
+            type: 'candlestick',
+            id: 'aapl',
+            name: 'AAPL',
+            data: data
+        }, {
+            type: 'column',
+            id: 'volume',
+            name: 'Volume',
+            data: volume,
+            yAxis: 1
+        }]
+    });
+
+    document.getElementById("overlays").addEventListener("change", function (e) {
+        const seriesToAdd = e.target.value;
+
+        // Destroy previously existing series.
+        if (chart.series[2] && chart.series[2].yAxis.options.index === 0) {
+            chart.series[2].destroy(false);
+        }
+        if (chart.series[3] && chart.series[3].yAxis.options.index === 0) {
+            chart.series[3].destroy(false);
+        }
```

Comment:
What is the purpose of this if statement?

Reply:
User can add first an oscillator (which will be series with the index 2), then can add an overlay (which will be with index 3). The only thing (as far as I can see) which you can use to distinguish them is the yAxis index.
The main purpose for that statements is to detect which series is assigned to which axis and destroy the correct one.
Is there any room for improvement?